tests/models/language.py

```diff
@@ -17,3 +20,143 @@ def test_language_repr():
     assert (
         "<Language: %s>" % language.name
         == repr(language))
+
+
+@pytest.mark.django_db
+def test_unused_language_is_not_live():
+    unused_language = LanguageDBFactory()
+    assert unused_language.translationproject_set.count() == 0
+    assert unused_language not in Language.live.all()
+
+
+@pytest.mark.django_db
+def test_language_only_in_disabled_proj_is_not_live():
+    language = LanguageDBFactory()
+    project = ProjectDBFactory(source_language=language)
+    project.disabled = True
+    project.save()
+    TranslationProjectFactory(language=language, project=project)
+
+    assert project.disabled
+    assert language.translationproject_set.count() == 1
+    assert language not in Language.live.all()
+
+
+@pytest.mark.django_db
+def test_language_only_in_obsolete_tp_is_not_live():
+    language = LanguageDBFactory()
+    tp = TranslationProjectFactory(
+        language=language,
+        project=ProjectDBFactory(source_language=language)
+    )
+    tp.directory.makeobsolete()
+
+    assert tp.directory.obsolete
+    assert language.translationproject_set.count() == 1
+    assert language not in Language.live.all()
+
+
+@pytest.mark.django_db
+def test_language_only_in_disabled_project_and_obsolete_tp_is_not_live():
+    language = LanguageDBFactory()
+
+    # Create TP for disabled project.
+    project = ProjectDBFactory(source_language=language)
+    project.disabled = True
+    project.save()
+    TranslationProjectFactory(language=language, project=project)
+
+    # Create obsolete TP.
+    tp = TranslationProjectFactory(
+        language=language,
+        project=ProjectDBFactory(source_language=language)
+    )
+    tp.directory.makeobsolete()
+
+    assert project.disabled
+    assert tp.directory.obsolete
+    assert language.translationproject_set.count() == 2
+    assert language not in Language.live.all()
+
+
+@pytest.mark.django_db
+def test_language_both_in_disabled_project_and_regular_tp_is_live():
+    language = LanguageDBFactory()
+
+    # Create TP for disabled project.
+    project = ProjectDBFactory(source_language=language)
+    project.disabled = True
+    project.save()
+    TranslationProjectFactory(language=language, project=project)
+
+    # Create regular TP.
+    TranslationProjectFactory(
+        language=language,
+        project=ProjectDBFactory(source_language=language)
+    )
+
+    assert project.disabled
+    assert language.translationproject_set.count() == 2
+    assert language in Language.live.all()
+
+
+@pytest.mark.django_db
+def test_language_both_in_obsolete_and_regular_tp_is_live():
+    language = LanguageDBFactory()
+
+    # Create obsolete TP.
+    language = LanguageDBFactory()
+    tp = TranslationProjectFactory(
```

Comment:
It fails too. Using `tp0` forces the test to use `language0`, but that is being used in several TPs and therefore we cannot ensure if it is fit for the test.